modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/source/MappingTransforms.java

```diff
@@ -8,10 +8,20 @@
 
 package org.elasticsearch.datastreams.logsdb.qa.matchers.source;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 class MappingTransforms {
+    /**
+     * Container for mapping of a field. Contain field mapping parameters and mapping parameters of parent fields (if present)
+     * in order of increasing distance (direct parent first).
```

Comment:
Nit: add a note here why we need to track all parent params (e.g. in case any has "enabled:false", source gets stored as is).